### PR TITLE
feature: traffic permissions

### PR DESCRIPTION
**Pull request by gszr**
- Match TrafficPermissions against dataplane
- Create RBAC configuration for matched TrafficPermissions
- Update tests for TrafficPermissions

### CODE REVIEW

File: pkg/core/permissions/matcher.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+func (m *TrafficPermissionsMatcher) Match(ctx context.Context, dataplane *mesh_core.DataplaneResource) (*mesh_core.TrafficPermissionResourceList, error) {
+	permissions := &mesh_core.TrafficPermissionResourceList{}
+	if err := m.ResourceManager.List(ctx, permissions, store.ListByMesh(dataplane.GetMeta().GetMesh())); err != nil {
+		return nil, err
+	}
+	return MatchDataplaneTrafficPermissions(&dataplane.Spec, permissions), nil
+}
+
+func MatchDataplaneTrafficPermissions(dataplane *mesh_proto.Dataplane, trafficPermissions *mesh_core.TrafficPermissionResourceList) *mesh_core.TrafficPermissionResourceList {
```
Comment (yskopets) [inline pkg/core/permissions/matcher.go]: We should return a list of matching `TrafficPermissions` for each listener individually.

Let's create an issue an fix it in a separate PR

File: pkg/xds/envoy/rbac.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+			}
+			principals = append(principals, principal)
+		}
+	}
+
+	return &rbac_config.Policy{
+		Permissions: []*rbac_config.Permission{
+			{
+				Rule: &rbac_config.Permission_Any{
+					// todo(jakubdyszkiewicz) for now it matches on any destination port, which means that
```
Comment (yskopets) [inline pkg/xds/envoy/rbac.go]: Let's create an issue for that

Comment (jakubdyszkiewicz) [reply pkg/xds/envoy/rbac.go]: Created https://github.com/Kong/kuma/issues/222

File: pkg/core/permissions/matcher.go (hunk 1 above)
Comment (jakubdyszkiewicz) [reply pkg/core/permissions/matcher.go]: Created https://github.com/Kong/kuma/issues/222

File: pkg/xds/envoy/rbac.go [hunk 3]
```diff
@@ -45,22 +46,27 @@ func createPolicy(permission *mesh_core.TrafficPermissionResource) *rbac_config.
...
+				principal.Identifier = &rbac_config.Principal_Any{
+					Any: true,
+				}
+			} else {
+				principal.Identifier = &rbac_config.Principal_Authenticated_{
 					Authenticated: &rbac_config.Principal_Authenticated{
 						PrincipalName: &matcher.StringMatcher{
 							MatchPattern: &matcher.StringMatcher_Exact{
-								Exact: "spiffe://" + permission.Meta.GetMesh() + "/" + source.Match["service"],
+								Exact: "spiffe://" + permission.Meta.GetMesh() + "/" + service,
```
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply pkg/xds/envoy/rbac.go]: Changed

**Pull request merged**